src/env/environment.rs

```diff
@@ -595,12 +632,33 @@
         .set(inherited_vars)
         .expect("env_init is being called multiple times");
 
+    // Set $USER, $HOME and $EUID
+    // This involves going to passwd and stuff.
+    vars.set_one(L!("EUID"), EnvMode::GLOBAL, geteuid().to_wstring());
+    setup_user(vars);
+
     if let Some(paths) = paths {
-        vars.set_one(
-            FISH_DATADIR_VAR,
-            EnvMode::GLOBAL,
-            str2wcstring(paths.data.as_os_str().as_bytes()),
-        );
+        let datadir = if cfg!(feature = "installable") {
+            if let Some(home) = vars.get(L!("HOME")) {
+                if let Ok(strhome) = std::str::from_utf8(&wcs2string(&home.as_string())) {
+                    let dir = PathBuf::from(strhome)
```

Comment:
I guess since we're on Unix we could also hardcode the slashes.
But going via PathBuf seems better. We have a convenience method for creating an `OsStr` which imlements `AsRef<Path>` which is required by `PathBuf::from`:

```diff
diff --git a/src/env/environment.rs b/src/env/environment.rs
index 041fc926b..00d49250d 100644
--- a/src/env/environment.rs
+++ b/src/env/environment.rs
@@ -642,12 +642,7 @@ pub fn env_init(paths: Option<&ConfigPaths>, do_uvars: bool, default_paths: bool
             if let Some(home) = vars.get(L!("HOME")) {
-                if let Ok(strhome) = std::str::from_utf8(&wcs2string(&home.as_string())) {
-                    let dir = PathBuf::from(strhome)
-                        .join(env!("DATADIR"))
-                        .join(env!("DATADIR_SUBDIR"));
-                    str2wcstring(dir.as_os_str().as_bytes())
-                } else {
-                    // This should be impossible given we already read_init.
-                    FLOG!(error, "Cannot decode $HOME");
-                    "".into()
-                }
+                let strhome = wcs2osstring(&home.as_string());
+                let dir = PathBuf::from(strhome)
+                    .join(env!("DATADIR"))
+                    .join(env!("DATADIR_SUBDIR"));
+                str2wcstring(dir.as_os_str().as_bytes())
             } else {
```